src/core/internal/lifetime.d

```diff
@@ -84,23 +84,87 @@ if (is(UT == core.internal.traits.Unqual!UT))
     emplaceRef!(UT, UT)(chunk, forward!args);
 }
 
-//emplace helper functions
-private nothrow pure @trusted
-void emplaceInitializer(T)(scope ref T chunk)
+/+
+Emplaces T.init.
+In contrast to `emplaceRef(chunk)`, there are no checks for disabled default
+constructors etc.
++/
+void emplaceInitializer(T)(scope ref T chunk) nothrow pure @trusted
+    if (!is(T == const) && !is(T == immutable) && !is(T == inout))
 {
-    // Emplace T.init.
-    // Previously, an immutable static and memcpy were used to hold an initializer.
-    // With improved unions, this is no longer needed.
-    union UntypedInit
+    import core.internal.traits : hasElaborateAssign;
+
+    static if (!hasElaborateAssign!T && __traits(compiles, chunk = T.init))
     {
-        T dummy;
+        chunk = T.init;
     }
-    static struct UntypedStorage
+    else
+    {
+        static if (__traits(isZeroInit, T))
```

Comment:
`else static if (__traits(isZeroInit, T))`

Reply:
I'd rather leave as-is; e.g., in case a `memset` is slower than the union version for DMD, one could add an `enum preferMemset = ...`.

Reply:
I meant to reduce nesting.

Reply:
[I know.]